src/libraries/System.Text.Json/tests/Serialization/ConstructorTests/ConstructorTests.Stream.cs

```diff
@@ -28,36 +28,31 @@ async Task RunTest<T>(byte[] testData)
-            tasks[1] = Task.Run(async () => await RunTest<ClassWithConstructor_SimpleAndComplexParameters>(ClassWithConstructor_SimpleAndComplexParameters.s_data));
-            tasks[2] = Task.Run(async () => await RunTest<Parameterized_Class_With_ComplexTuple>(Parameterized_Class_With_ComplexTuple.s_data));
+            await RunTest<ClassWithConstructor_SimpleAndComplexParameters>(ClassWithConstructor_SimpleAndComplexParameters.s_data);
+            await RunTest<Parameterized_Class_With_ComplexTuple>(Parameterized_Class_With_ComplexTuple.s_data);
             // JSON that doesn't bind to ctor args are matched with properties or ignored (as appropriate).
-            tasks[3] = Task.Run(async () => await RunTest<Person_Class>(Person_Class.s_data));
-            tasks[4] = Task.Run(async () => await RunTest<Person_Struct>(Person_Struct.s_data));
+            await RunTest<Person_Class>(Person_Class.s_data);
+            await RunTest<Person_Struct>(Person_Struct.s_data);
             // JSON that doesn't bind to ctor args or properties are sent to ext data if avaiable.
-            tasks[5] = Task.Run(async () => await RunTest<Parameterized_Person>(Parameterized_Person.s_data));
-            tasks[6] = Task.Run(async () => await RunTest<Parameterized_Person_ObjExtData>(Parameterized_Person_ObjExtData.s_data));
+            await RunTest<Parameterized_Person>(Parameterized_Person.s_data);
+            await RunTest<Parameterized_Person_ObjExtData>(Parameterized_Person_ObjExtData.s_data);
             // Up to 64 ctor args are supported.
-            tasks[7] = Task.Run(async () => await RunTest<Class_With_Ctor_With_64_Params>(Class_With_Ctor_With_64_Params.Data));
+            await RunTest<Class_With_Ctor_With_64_Params>(Class_With_Ctor_With_64_Params.Data);
             // Arg deserialization honors attributes on matching property.
-            tasks[8] = Task.Run(async () => await RunTest<Point_MembersHave_JsonPropertyName>(Point_MembersHave_JsonPropertyName.s_data));
-            tasks[9] = Task.Run(async () => await RunTest<Point_MembersHave_JsonConverter>(Point_MembersHave_JsonConverter.s_data));
-            tasks[10] = Task.Run(async () => await RunTest<Point_MembersHave_JsonIgnore>(Point_MembersHave_JsonIgnore.s_data));
+            await RunTest<Point_MembersHave_JsonPropertyName>(Point_MembersHave_JsonPropertyName.s_data);
+            await RunTest<Point_MembersHave_JsonConverter>(Point_MembersHave_JsonConverter.s_data);
+            await RunTest<Point_MembersHave_JsonIgnore>(Point_MembersHave_JsonIgnore.s_data);
             // Complex JSON as last argument works
-            tasks[11] = Task.Run(async () => await RunTest<Point_With_Array>(Point_With_Array.s_data));
-            tasks[12] = Task.Run(async () => await RunTest<Point_With_Dictionary>(Point_With_Dictionary.s_data));
-            tasks[13] = Task.Run(async () => await RunTest<Point_With_Object>(Point_With_Object.s_data));
-
-            Task.WaitAll(tasks);
```

Comment:
Can you also apply the pattern suggested in https://github.com/dotnet/runtime/pull/38663/files#r452481795 here?

Reply:
Isn't the intent of this test to detect race conditions? By awaiting one-by-one doesn't defy said purpose?